openshift_cli_installer/libs/clusters/ipi_cluster.py

```diff
@@ -105,27 +112,53 @@ def _create_install_config_file(self):
         if fips:
             terraform_parameters["fips"] = fips
 
-        cluster_install_config = get_install_config_j2_template(jinja_dict=terraform_parameters)
+        cluster_install_config = get_install_config_j2_template(jinja_dict=terraform_parameters, platform=platform)
 
         with open(os.path.join(self.cluster_info["cluster-dir"], "install-config.yaml"), "w") as fd:
             fd.write(yaml.dump(cluster_install_config))
 
     def _set_install_version_url(self):
         cluster_version = self.cluster["version"]
-        version_url = [url for url, versions in self.aws_base_available_versions.items() if cluster_version in versions]
+        version_url = [url for url, versions in self.ipi_base_available_versions.items() if cluster_version in versions]
         if version_url:
             self.cluster_info["version-url"] = f"{version_url[0]}:{self.cluster_info['version']}"
         else:
             self.logger.error(
                 f"{self.log_prefix}: Cluster version url not found for"
-                f" {cluster_version} in {self.aws_base_available_versions.keys()}",
+                f" {cluster_version} in {self.ipi_base_available_versions.keys()}",
             )
             raise click.Abort()
 
+    def get_gcp_project_id(self):
+        with open(self.gcp_service_account_file, "r") as fd:
+            gcp_sa_file_content = json.load(fd)
+
+        return gcp_sa_file_content["project_id"]
+
+    def move_gcp_service_account_file(self, action):
```

Comment:
should be done for `create` or `destroy` separately - i.e if the file exists, use it during the flow and restore once the cluster is created/destroyed.

Reply:
for this function, `action` param is used for create or restore `~/.gcp/osServiceAccount.json` file.
This function is called each time before and after running installer command.

Reply:
using `action` here is misleading and is not tied to create / destroy
please rename